### PR TITLE
Add validation preloaded_pt field in PT_UPDATE type of table

### PR DESCRIPTION
**Pull request by Kozoriz**
In case when we have type PT_UPDATE of policy_table 
and preloaded_pt field in present in PT. 
We must invalidate this table. 

Fixed defect : APPLINK-30866

### CODE REVIEW

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 1]
```diff
@@ -704,22 +704,37 @@ void ModuleConfig::ReportErrors(rpc::ValidationReport* report__) const {
...
-    rpc::ValidationReport* ommited_field_report;
-    if (vehicle_make.is_initialized()) {
-      ommited_field_report = &report__->ReportSubobject("vehicle_make");
-      ommited_field_report->set_validation_info(validation_info);
-    }
-    if (vehicle_year.is_initialized()) {
-      ommited_field_report = &report__->ReportSubobject("vehicle_year");
-      ommited_field_report->set_validation_info(validation_info);
+
+  std::string validation_info =
```
Comment (dev-gh) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @Kozoriz 
const ?

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 2]
```diff
@@ -704,22 +704,37 @@ void ModuleConfig::ReportErrors(rpc::ValidationReport* report__) const {
...
+      ommited_validation_info + PolicyTableTypeToString(GetPolicyTableType());
```
Comment (dev-gh) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @Kozoriz 
omitted

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 3]
```diff
@@ -704,22 +704,37 @@ void ModuleConfig::ReportErrors(rpc::ValidationReport* report__) const {
...
+  rpc::ValidationReport* ommited_field_report;
```
Comment (dev-gh) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @Kozoriz 
Init value

Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @Kozoriz I can't see why you need this value out of switch case scope, also why it is a pointer, you can use reference.

File: src/components/policy/policy_external/src/policy_table/validation.cc [hunk 4]
```diff
@@ -148,16 +148,26 @@ bool Rpcs::Validate() const {
...
+        return false;
+      }
+      break;
     }
-    if (vehicle_model.is_initialized()) {
-      return false;
+    case PT_UPDATE: {
+      if (preloaded_pt->is_initialized()) {
+        return false;
+      }
```
Comment (dcherniev) [inline src/components/policy/policy_external/src/policy_table/validation.cc]: @Kozoriz Please add missing "break;" here.

File: src/components/policy/policy_external/src/policy_table/validation.cc [hunk 5]
```diff
@@ -148,16 +148,26 @@ bool Rpcs::Validate() const {
...
 bool ModuleConfig::Validate() const {
-  if (PT_PRELOADED == GetPolicyTableType()) {
-    if (vehicle_make.is_initialized()) {
-      return false;
-    }
-    if (vehicle_year.is_initialized()) {
-      return false;
+  switch (GetPolicyTableType()) {
+    case PT_PRELOADED: {
+      if (vehicle_make.is_initialized()) {
```
Comment (LuxoftAKutsan) [inline src/components/policy/policy_external/src/policy_table/validation.cc]: @Kozoriz you can use ```||``` or helpers.

**Comment by Kozoriz** [comment]
@dev-gh , @LuxoftAKutsan , @dcherniev , all fixed. PLease review

**Comment by ovvasyliev** [comment]
Rebuild required

**Pull request merged**